bundles/org.openhab.binding.mikrotik/src/main/java/org/openhab/binding/mikrotik/internal/handler/MikrotikRouterosBridgeHandler.java

```diff
@@ -0,0 +1,271 @@
+                }
+                updateStatus(ThingStatus.ONLINE);
+            } catch (MikrotikApiException e) {
+                logger.warn("Error while logging in to RouterOS {}", getThing().getUID(), e);
+                logger.debug("RouterOS connection exception cause", e.getCause());
+
+                if (e.getMessage().contains("Command timed out") || e.getMessage().contains("Error connecting")) {
+                    routeros.stop();
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                } else if (e.getMessage().contains("Connection refused")) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "Remote host refused to connect, make sure port is correct");
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+                }
+            }
+        } else {
+            // We're connected - do a usual polling cycle
+            performRefresh();
+        }
+    }
+
+    private void performRefresh() {
+        try {
+            logger.debug("Refreshing RouterOS caches for {}", getThing().getUID());
+            routeros.refresh();
+            // refresh own channels
+            for (Channel channel : getThing().getChannels()) {
+                try {
+                    refreshChannel(channel.getUID());
+                } catch (RuntimeException e) {
+                    throw new ChannelUpdateException(getThing().getUID(), channel.getUID(), e);
+                }
+            }
+            // refresh all the client things below
+            getThing().getThings().forEach(thing -> {
+                ThingHandler handler = thing.getHandler();
+                if (handler instanceof MikrotikBaseThingHandler) {
+                    ((MikrotikBaseThingHandler) handler).refresh();
+                }
+            });
+        } catch (ChannelUpdateException e) {
+            logger.error("Error updating channel! {}", e.getMessage(), e.getCause());
+        } catch (MikrotikApiException e) {
+            logger.error("Failed to refresh RouterOS cache in {}", getThing().getUID(), e);
+            routeros.stop();
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (Exception e) {
+            logger.error("Unhandled exception while refreshing the {} RouterOS model", getThing().getUID(), e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Handling command = {} for channel = {}", command, channelUID);
+        if (getThing().getStatus() == ONLINE) {
+            @Nullable
+            RouterosDevice routeros = getRouteros();
```

Comment:
```suggestion
            RouterOsDevice routerOs = getRouterOs();
```

Reply:
Can you comment on this?

Reply:
RouterOS = Router Operating System
But I'm still not sure which casing for the code is better 🤔

Reply:
You can leave it as is.